agent/api/task/task.go

```diff
@@ -2338,59 +2338,147 @@ func (task *Task) dockerLinks(container *apicontainer.Container, dockerContainer
 }
 
 var getHostPortRange = utils.GetHostPortRange
+var getHostPort = utils.GetHostPort
+
+// In buildPortMapWithSCIngressConfig, the dockerPortMap and the containerPortSet will be constructed
+// for ingress listeners under two service connect bridge mode cases:
+// (1) non-default bridge mode service connect experience: customers specify host ports for listeners in the ingress config.
+// (2) default bridge mode service connect experience: customers do not specify host ports for listeners in the ingress config.
+// Instead, ECS Agent finds host ports within the given dynamic host port range.
+// An error will be returned for case (2) if ECS Agent cannot find an available host port within range.
+func (task *Task) buildPortMapWithSCIngressConfig(dynamicHostPortRange string) (nat.PortMap, map[int]struct{}, error) {
+	var err error
+	ingressDockerPortMap := nat.PortMap{}
+	ingressContainerPortSet := make(map[int]struct{})
+	protocolStr := "tcp"
+	for _, ic := range task.ServiceConnectConfig.IngressConfig {
+		listenerPortInt := int(ic.ListenerPort)
+		dockerPort := nat.Port(strconv.Itoa(listenerPortInt) + "/" + protocolStr)
+		hostPortStr := ""
+		if ic.HostPort != nil {
+			// For non-default bridge mode service connect experience, a host port is specified by customers
+			hostPortStr = strconv.Itoa(int(*ic.HostPort))
```

Comment:
where are we validating the ic.HostPort input is a valid integer?

Reply:
The service connect config is streamed down from ACS to ECS Agent as a task attachment, and ECS Agent validate the whole service connect config when receiving a task payload message from ACS.   

See the workflow listed as follows:
1. ECS Agent converts and validate task attachments streamed down from ACS in [TaskFromACS()](https://github.com/aws/amazon-ecs-agent/blob/master/agent/api/task/task.go#L333)
2. ECS Agent parses the service connect config from the task attachment in [handleTaskAttachments()](https://github.com/aws/amazon-ecs-agent/blob/4943e4d853a84cf4d76b9d1241ee1f4f189b9c4a/agent/api/task/task_attachment_handler.go#L97)
3. ECS Agent validate the parsed service connect config in [handleTaskAttachments()](https://github.com/aws/amazon-ecs-agent/blob/4943e4d853a84cf4d76b9d1241ee1f4f189b9c4a/agent/api/task/task_attachment_handler.go#L102)
4. Service connect ingress config will be validated in [service_connect_validator.go](https://github.com/aws/amazon-ecs-agent/blob/4943e4d853a84cf4d76b9d1241ee1f4f189b9c4a/agent/api/serviceconnect/service_connect_validator.go#L310), and host ports will be validated [here](https://github.com/aws/amazon-ecs-agent/blob/4943e4d853a84cf4d76b9d1241ee1f4f189b9c4a/agent/api/serviceconnect/service_connect_validator.go#L226) to ensure host ports are within a valid port range (1 to 65535) and do not have port collisions.

Based on this, we don't perform additional validations in function `buildPortMapWithSCIngressConfig()`.